### PR TITLE
[docs][DOC-64] Auto tablet splitting clarification

### PR DESCRIPTION
**Pull request by ddhodge**
Update automatic tablet splitting description for 2.18

preview/reference/configuration/yb-tserver/#yb-num-shards-per-tserver

preview/architecture/docdb-sharding/tablet-splitting/#automatic-tablet-splitting

preview/explore/linear-scalability/sharding-data/

@netlify /preview/architecture/docdb-sharding/tablet-splitting/

### CODE REVIEW

File: docs/content/preview/architecture/docdb-sharding/tablet-splitting.md [hunk 1]
```diff
@@ -215,13 +215,20 @@ For details on the architecture design, see [Automatic resharding of data with t
...
+When automatic tablet splitting is enabled, newly-created tables with [hash sharding](../../../architecture/docdb-sharding/sharding/#hash-sharding) have one tablet per node by default.
 
-When automatic tablet splitting is enabled, newly-created tables with [hash sharding](../../../architecture/docdb-sharding/sharding/#hash-sharding) have one shard **per _yb-tserver_** by default; from version `2.14.10` such tables have one shard (for servers with up to 2 CPU cores) or two shards (for servers with up to 4 CPU cores) **per _cluster_** by default.
+The following additional defaults apply:
+
+* From version 2.14.10, for servers with up to 2 CPU cores, newly-created tables have one tablet or, for servers with up to 4 CPU cores, two tablets **per _cluster_** by default.
+* From version 2.18, for servers with up to 2 CPU cores, newly-created tables have one tablet per table, and servers with up to 4 CPU cores have 2 tablets per table.
```
Comment (arybochkin) [inline docs/content/preview/architecture/docdb-sharding/tablet-splitting.md:223]: These lines describe the same behaviour, you should keep only one line. The behaviour was changed starting form 2.14.10.

File: docs/content/preview/architecture/docdb-sharding/tablet-splitting.md [hunk 2]
```diff
@@ -215,13 +215,20 @@ For details on the architecture design, see [Automatic resharding of data with t
...
+
+From version 2.18.0, automatic tablet splitting is turned on by default.
+
+To control automatic tablet splitting, use the `yb-master` [`--enable_automatic_tablet_splitting`](../../../reference/configuration/yb-master/#enable-automatic-tablet-splitting) flag and specify the associated flags to configure when tablets should split, and use `yb-tserver` [`--enable_automatic_tablet_splitting`](../../../reference/configuration/yb-tserver/#enable-automatic-tablet-splitting). The flags must match on all `yb-master` and `yb-tserver` configurations of a YugabyteDB cluster.
 
 {{< note title="Note" >}}
 
-Newly-created tables with [range sharding](../../../architecture/docdb-sharding/sharding/#range-sharding) always have one shard *per cluster* unless table partitioning is specified explicitly during table creation.
+Newly-created tables with [range sharding](../../../architecture/docdb-sharding/sharding/#range-sharding) always have one shard _per cluster_ unless table partitioning is specified explicitly during table creation.
```
Comment (arybochkin) [inline docs/content/preview/architecture/docdb-sharding/tablet-splitting.md:231]: Maybe `one shard` should be replaced with `one tablet` to follow the style which is being used above in the changes.

File: docs/content/preview/reference/configuration/yugabyted.md [hunk 3]
```diff
@@ -23,18 +23,6 @@ For examples of using yugabyted to deploy single- and multi-node clusters, see [
...
-macOS Monterey enables AirPlay receiving by default, which listens on port 7000. This conflicts with YugabyteDB and causes `yugabyted start` to fail. Use the [--master_webserver_port flag](#advanced-flags) when you start the cluster to change the default port number, as follows:
-
-```sh
-./bin/yugabyted start --master_webserver_port=9999
-```
-
-Alternatively, you can disable AirPlay receiving, then start YugabyteDB normally, and then, optionally, re-enable AirPlay receiving.
-
-{{% /note %}}
-
```
Comment (arybochkin) [inline docs/content/preview/reference/configuration/yugabyted.md:37]: This change is not related to automatic tablet splitting, why is it removed in the context of the update?

File: docs/content/preview/reference/configuration/yugabyted.md [hunk 4]
```diff
@@ -681,6 +669,27 @@ The following are combinations of environment variables and their uses:
...
+
+On macOS, every additional node after the first needs a loopback address configured to simulate the use of multiple hosts or nodes. For example, for a three-node cluster, you add two additional addresses as follows:
+
+```sh
+sudo ifconfig lo0 alias 127.0.0.2
+sudo ifconfig lo0 alias 127.0.0.3
+```
+
+The loopback addresses do not persist upon rebooting your computer.
+
```
Comment (arybochkin) [inline docs/content/preview/reference/configuration/yugabyted.md:692]: These changes are outside the automatic tablet splitting clarification. Do we need them in this pull request? Can a separate PR be used?

File: docs/content/preview/reference/configuration/yb-tserver.md
Comment (arybochkin) [inline docs/content/preview/reference/configuration/yb-tserver.md:1]: It's good to update --ysql_num_shards_per_tserver also.

File: docs/content/preview/architecture/docdb-sharding/tablet-splitting.md (hunk 1 above)
Comment (ddhodge) [reply docs/content/preview/architecture/docdb-sharding/tablet-splitting.md:223]: Oh, so this started with 2.14.10, not 2.18, it is only the setting of the default that changed with 2.18. OK that clarifies things

File: docs/content/preview/architecture/docdb-sharding/tablet-splitting.md (hunk 2 above)
Comment (ddhodge) [reply docs/content/preview/architecture/docdb-sharding/tablet-splitting.md:231]: Also, sometimes it is 'per cluster' other times it is 'per table' - but (from previous comment) these appear to be equivalent, so will use 'per table'

File: docs/content/preview/reference/configuration/yugabyted.md (hunk 4 above)
Comment (ddhodge) [reply docs/content/preview/reference/configuration/yugabyted.md:692]: Sure OK. It was a peripheral change due to instructions for creating cluster for example on the sharding page.

File: docs/content/preview/reference/configuration/yb-tserver.md
Comment (ddhodge) [reply docs/content/preview/reference/configuration/yb-tserver.md:1]: Yes, will update other flags and port to stable when we're good with the changes

File: docs/content/preview/architecture/docdb-sharding/tablet-splitting.md [hunk 5]
```diff
@@ -215,13 +215,17 @@ For details on the architecture design, see [Automatic resharding of data with t
...
+When automatic tablet splitting is enabled, newly-created tables with [hash sharding](../../../architecture/docdb-sharding/sharding/#hash-sharding) have one tablet per node by default.
 
-When automatic tablet splitting is enabled, newly-created tables with [hash sharding](../../../architecture/docdb-sharding/sharding/#hash-sharding) have one shard **per _yb-tserver_** by default; from version `2.14.10` such tables have one shard (for servers with up to 2 CPU cores) or two shards (for servers with up to 4 CPU cores) **per _cluster_** by default.
+In addition, from version 2.14.10, for servers with up to 2 CPU cores, newly-created tables have one tablet per table, and servers with up to 4 CPU cores have 2 tablets per table.
```
Comment (druzac) [inline docs/content/preview/architecture/docdb-sharding/tablet-splitting.md:220]: "newly-created tables have one tablet per table"
It's strange we're expressing the number of tablets in terms of the number of tables while talking about a single table creation. How about:

For servers with up to 2 CPU cores, newly-created tables will have a single tablet. For servers with up to 4 CPU cores, newly-created tables will have 2 tablets.

File: docs/content/preview/architecture/docdb-sharding/tablet-splitting.md [hunk 6]
```diff
@@ -215,13 +215,17 @@ For details on the architecture design, see [Automatic resharding of data with t
...
+
+From version 2.18.0, automatic tablet splitting is turned on by default.
+
+To control automatic tablet splitting, use the `yb-master` [`--enable_automatic_tablet_splitting`](../../../reference/configuration/yb-master/#enable-automatic-tablet-splitting) flag and specify the associated flags to configure when tablets should split, and use `yb-tserver` [`--enable_automatic_tablet_splitting`](../../../reference/configuration/yb-tserver/#enable-automatic-tablet-splitting). The flags must match on all `yb-master` and `yb-tserver` configurations of a YugabyteDB cluster.
 
 {{< note title="Note" >}}
 
-Newly-created tables with [range sharding](../../../architecture/docdb-sharding/sharding/#range-sharding) always have one shard *per cluster* unless table partitioning is specified explicitly during table creation.
+Newly-created tables with [range sharding](../../../architecture/docdb-sharding/sharding/#range-sharding) always have one tablet _per table_ unless table partitioning is specified explicitly during table creation.
```
Comment (druzac) [inline docs/content/preview/architecture/docdb-sharding/tablet-splitting.md:228]: Likewise, can we say newly created tables will have a single tablet?

**Review by druzac — APPROVED** [review]

File: docs/content/preview/architecture/docdb-sharding/tablet-splitting.md (hunk 5 above)
Comment (ddhodge) [reply docs/content/preview/architecture/docdb-sharding/tablet-splitting.md:220]: I think it is to make clear the distinction that it is NOT per node (previous paragraph). Because the previous paragraph specified 1 tablet per node, it may be interpreted ambiguously without per table here. @arybochkin

Comment (arybochkin) [reply docs/content/preview/architecture/docdb-sharding/tablet-splitting.md:220]: That's why _per cluster_ was used previously..

File: docs/content/preview/architecture/docdb-sharding/tablet-splitting.md (hunk 1 above)
Comment (arybochkin) [reply docs/content/preview/architecture/docdb-sharding/tablet-splitting.md]: It was initially implemented for 2.19 (master) and then backported to 2.18, 2.16 and 2.14. Probably it is OK to have 2.14.10. But originally each docs version mentioned the corresponding backport version: for 2.14 it was 2.14.10, for 2.16 - 2.16.5 and so on.

**Comment by arybochkin** [comment]
@ddhodge , just to check, are you going to update `stable`, `2.18`, `2.16` and `2.14` as well?

**Review by arybochkin — APPROVED** [review]

File: docs/content/preview/architecture/docdb-sharding/tablet-splitting.md (hunk 5 above)
Comment (ddhodge) [reply docs/content/preview/architecture/docdb-sharding/tablet-splitting.md:220]: We can standardize on per cluster if you like - there was a mix before

**Comment by aishwarya24** [comment]
The changes also need to be updated in [yb-master](https://deploy-preview-18088--infallible-bardeen-164bc9.netlify.app/preview/reference/configuration/yb-master/#yb-num-shards-per-tserver) reference section.

**Review by aishwarya24 — APPROVED** [review]

File: docs/content/preview/architecture/docdb-sharding/tablet-splitting.md (hunk 5 above)
Comment (arybochkin) [reply docs/content/preview/architecture/docdb-sharding/tablet-splitting.md]: I'm ok to go with proposed changes.
